src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -475,19 +483,32 @@ def _handleOnlyStateTransition(self, workload, request_args, dn):
         It handles only the state transition.
         Special handling needed if a request is aborted or force completed.
         """
+        # if we got here, then the main workflow has been already validated
+        # and the status transition is allowed
         req_status = request_args["RequestStatus"]
         cascade = request_args.get("cascade", False)
 
         if req_status in ["aborted", "force-complete"]:
             # cancel the workflow first
             self.gq_service.cancelWorkflow(workload.name())
-        if req_status in ["rejected", "closed-out", "announced"] and cascade:
-            cascade_list = self._retrieveResubmissionChildren(workload.name())
-            for req_name in cascade_list:
-                self.reqmgr_db_service.updateRequestStatus(req_name, req_status, dn)
 
-        cherrypy.log('Updating request status for request "%s" to %s. Cascade mode: %s' % (workload.name(), req_status, cascade))
-        # then update original workflow status in couchdb
+        # cascade option is only supported for these 3 statuses. If set, we need to
+        # find all the children requests and perform the same status transition
+        if req_status in ["rejected", "closed-out", "announced"] and cascade:
+            childrenNamesAndStatus = self._retrieveResubmissionChildren(workload.name())
+            msg = "Workflow {} has {} ".format(workload.name(), len(childrenNamesAndStatus))
+            msg += "children workflows to have a status transition to: {}".format(req_status)
+            cherrypy.log(msg)
+            for childInfo in childrenNamesAndStatus:
+                if check_allowed_transition(childInfo['value'], req_status):
+                    cherrypy.log('Updating request status for {} to {}.'.format(childInfo['id'], req_status))
+                    self.reqmgr_db_service.updateRequestStatus(childInfo['id'], req_status, dn)
+                else:
+                    msg = "Status transition from {} to {} ".format(childInfo['value'], req_status)
+                    msg += "not allowed for workflow: {}, skippint it!".format(childInfo['id'])
+                    cherrypy.log(msg)
+        # then update the original/parent workflow status in couchdb
+        cherrypy.log('Updating request status for {} to {}.'.format(workload.name(), req_status))
         report = self.reqmgr_db_service.updateRequestStatus(workload.name(), req_status, dn)
```

Comment:
I know this line is not part of this commit, but just a a general comment - I do not like the attitude of triggering actions during assignments. BTW I am aware nothing can be done here in this regard, so this comment is not a request for a change.

Reply:
Don't see it as triggering an action, but as calling a function and retrieving the returned response ;)